test/integration/noderesourcetopology_test.go

```diff
@@ -839,7 +856,286 @@ func TestTopologyMatchPlugin(t *testing.T) {
 			},
 			expectedNodes: []string{},
 		},
-	} {
+	}
+
+	// a dedicated set of tests for the scope=container flow
+	scopeEqualsContainerTests := []nrtTestUserEntry{
+		{
+			description: "[1][tier3] single container with good allocation - fit",
```

Comment:
this is nice. My only (but not tiny) gripe is that both the expected outcome and most notably the NRT data is buried in the parsing helper, which requires back and forth from/out the helper to get the full test context.
I'm not sure how a good solution could look like though. Albeit quite uncomfortable, is not a blocker for this PR though.

Reply:
Additionally, I think integration test should cover only tier1 (as identified in `TESTS.md` in https://github.com/kubernetes-sigs/scheduler-plugins/pull/371). Otherwise this becomes unwieldly.
- unit tests are the cheapest, so they should give the most coverage
- integration tests should focus on the most important flows (at least initially)